client/ethereum.go

```diff
@@ -237,11 +272,15 @@ func (e *EthereumClient) WaitForTransaction(transactionHash common.Hash) error {
 	timeout := e.Network.Config().Timeout
 	confirmations := 0
 
+	ticker := time.NewTicker(200 * time.Millisecond)
+	defer ticker.Stop()
+
 	for {
 		select {
 		case err := <-subscription.Err():
 			return err
 		case header := <-headerChannel:
+			// FIXME: investigate why there is no new header with hardhat mining
```

Comment:
Just make sure this is fixed prior to merge.

Reply:
Tried to refactor it using background goroutine and concurrent map, it seems subscription is completely not working in hardhat for now because of https://github.com/nomiclabs/hardhat/pull/1163
Also tried it with geth - it's working, maybe we can use poll for now, and add required changes when we add geth?